stdlib/public/TensorFlow/ShapedArray.swift

```diff
@@ -485,7 +645,8 @@ extension ShapedArray : RandomAccessCollection, MutableCollection {
       precondition(!isScalar,
                    "Scalar has no elements and cannot be subscripted.")
       precondition(
-        indices ~= bounds.lowerBound && indices ~= bounds.upperBound - 1,
+        bounds.lowerBound >= startIndex && bounds.lowerBound <= endIndex &&
```

Comment:
The old condition led to crashes. For example, `indices = 0..<2` and `bounds = 2..<2` would crash.
The new logic is copied from `Array.subscript(bounds: Range<Int>)` and is correct (does not crash).